### PR TITLE
(SP: 1) Category model validation #59

### LINKED ISSUE #59: (SP: 1) Category model validation
## Description
There should be the following validation rules for Category model:
- **name** must be required
- **appearance.icon** must be required
- **appearance.color** must be required
## Acceptance Criteria
- [ ] all of the rules above must be added

### PR DESCRIPTION
**Pull request by ohanovdmytro**
close #59 

Added the following validation rules for Category model:

- [x] name must be required
- [x] appearance.icon must be required
- [x] appearance.color must be required

image

image

### CODE REVIEW

File: src/models/category.js
Comment (Olenka-Hryk) [inline src/models/category.js:1]: I think you should specify the type of error for the user, such as 'FIELD_IS_NOT_DEFINED'.

File: src/models/category.js [hunk 1]
```diff
@@ -5,15 +5,18 @@ const categorySchema = new Schema(
   {
     name: {
       type: String,
+      required: true,
```
Comment (Olenka-Hryk) [inline src/models/category.js:8]: required: [true, errors.FIELD_IS_NOT_DEFINED('name').message]

Comment (Olenka-Hryk) [inline src/models/category.js:8]:
```suggestion
      required: [true, errors.FIELD_IS_NOT_DEFINED('name').message]
```

File: src/models/category.js [hunk 2]
```diff
@@ -5,15 +5,18 @@ const categorySchema = new Schema(
...
       unique: true
     },
     appearance: {
       icon: {
         type: String,
+        required: true,
```
Comment (Olenka-Hryk) [inline src/models/category.js:14]:
```suggestion
        required: [true, errors.FIELD_IS_NOT_DEFINED('appearance.icon').message]
```

File: src/models/category.js
Comment (Olenka-Hryk) [inline src/models/category.js:1]: `const errors = require('~/consts/errors')`

File: src/models/category.js [hunk 3]
```diff
@@ -5,15 +5,18 @@ const categorySchema = new Schema(
...
         default: 'mocked-path-to-icon'
       },
       color: {
         type: String,
+        required: true,
```
Comment (Olenka-Hryk) [inline src/models/category.js:19]:
```suggestion
        required: [true, errors.FIELD_IS_NOT_DEFINED('color').message],
```

**Comment by Olenka-Hryk** [comment]
I also advise you not to add code photos to the pull request, as the code is already visible in the changes. Also, the screenshot from Mongo doesn't provide information on whether everything is working correctly. It would be cool to add screens of validation in Swagger/Postman and check if the validation of necessary fields is indeed performed.

File: src/models/category.js
Comment (Olenka-Hryk) [inline src/models/category.js:1]: You forget to add `const errors = require('~/consts/errors')` at the top of this file :)

**Review by docia — APPROVED** [review]

**Review by m6xcom — APPROVED** [review]

**Review by BodiaPutiak — APPROVED** [review]

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**